### PR TITLE
Refactor webdriverio utils interception devtools to TS

### PR DESCRIPTION
**Pull request by Fabianopb**
## Proposed changes

Refactor webdriverio utils interception devtools to TS

## Types of changes

- [x] TypeScript refactor

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) doc

### CODE REVIEW

File: packages/webdriverio/src/utils/interception/devtools.ts [hunk 1]
```diff
@@ -266,15 +298,15 @@ const canAccess = async (filepath) => {
...
         return JSON.parse(body) || body
     } catch {
         return body
     }
 }
 
-const logFetchError = (err) => {
+const logFetchError = (err?: Error) => {
     /* istanbul ignore next */
     log.debug(err && err.message ? err.message : err)
```
Comment (christian-bromann) [inline packages/webdriverio/src/utils/interception/devtools.ts:311]:
```suggestion
    log.debug(err?.message)
```

File: packages/webdriverio/src/utils/interception/devtools.ts [hunk 2]
```diff
@@ -207,12 +239,12 @@ export default class DevtoolsInterception extends Interception {
...
      * @param {string} errorReason  error code of the response
      */
-    abortOnce (errorReason) {
+    abortOnce (errorReason: string) {
         this.abort(errorReason, false)
     }
 }
 
-const filterMethod = (method, expected) => {
+const filterMethod = (method: string, expected?: ((method: string) => boolean) | string) => {
```
Comment (christian-bromann) [inline packages/webdriverio/src/utils/interception/devtools.ts:247]: We can create an expected type that looks like this:

```ts
type ExpectParameter<T> = ((param: T) => boolean) | T
```

and then use it throughout the file like this:
```suggestion
const filterMethod = (method: string, expected?: ExpectParameter<string>) => {
```

**Review by christian-bromann — CHANGES_REQUESTED** [review]

File: packages/webdriverio/src/utils/interception/devtools.ts (hunk 1 above)
Comment (Fabianopb) [reply packages/webdriverio/src/utils/interception/devtools.ts]: Yeah, better like that, changed!

File: packages/webdriverio/src/utils/interception/devtools.ts (hunk 2 above)
Comment (Fabianopb) [reply packages/webdriverio/src/utils/interception/devtools.ts]: Good suggestion! 👍 Changed!

**Review by christian-bromann — APPROVED** [review]
👍

**Pull request merged**